From: Will Deacon <will.deacon@arm.com>
Date: Mon, 11 Aug 2014 14:24:47 +0100
Subject: asm-generic: add memfd_create system call to unistd.h
Origin: https://git.kernel.org/linus/503e6636b6f96056210062be703356f4253b6db9
Bug-Debian: https://bugs.debian.org/760702

Commit 9183df25fe7b ("shm: add memfd_create() syscall") added a new
system call (memfd_create) but didn't update the asm-generic unistd
header.

This patch adds the new system call to the asm-generic version of
unistd.h so that it can be used by architectures such as arm64.

Cc: Arnd Bergmann <arnd@arndb.de>
Reviewed-by: David Herrmann <dh.herrmann@gmail.com>
Signed-off-by: Will Deacon <will.deacon@arm.com>
[bwh: Backported to 3.16:
 - Adjust context
 - Insert unimplemented-syscall entries for seccomp and getrandom]
---
--- a/include/uapi/asm-generic/unistd.h
+++ b/include/uapi/asm-generic/unistd.h
@@ -699,9 +699,13 @@ __SYSCALL(__NR_sched_setattr, sys_sched_
 __SYSCALL(__NR_sched_getattr, sys_sched_getattr)
 #define __NR_renameat2 276
 __SYSCALL(__NR_renameat2, sys_renameat2)
+__SYSCALL(277, sys_ni_syscall)
+__SYSCALL(278, sys_ni_syscall)
+#define __NR_memfd_create 279
+__SYSCALL(__NR_memfd_create, sys_memfd_create)
 
 #undef __NR_syscalls
-#define __NR_syscalls 277
+#define __NR_syscalls 280
 
 /*
  * All syscalls below here should go away really,
